src/table.rs

```diff
@@ -62,7 +64,7 @@ pub(crate) struct Page<T: Slot> {
 
     /// Vector with data. This is always created with the capacity/length of `PAGE_LEN`
     /// and uninitialized data. As we initialize new entries, we increment `allocated`.
-    data: Vec<UnsafeCell<T>>,
+    data: Box<[UnsafeCell<MaybeUninit<T>>; PAGE_LEN]>,
```

Comment:
We could drop the `Box` here entirely if we wanted as a `Page` is currently always allocated as a `Box<dyn>` anyways, but we'd need to be careful in that we never put a `Page` on the stack when we construct it then which is rather tricky in today's Rust